--- rtl8188fu-2019-10-05/include/autoconf.h.orig	2019-10-18 01:02:02.000000000 +0300
+++ rtl8188fu-2019-10-05/include/autoconf.h	2019-10-18 01:02:56.129468483 +0300
@@ -282,7 +282,7 @@
  */
 #define CONFIG_DEBUG /* DBG_871X, etc... */
 #ifdef CONFIG_DEBUG
-	#define DBG	1	/* for ODM & BTCOEX debug */
+	#define DBG	0	/* for ODM & BTCOEX debug */
 	#define DBG_PHYDM_MORE 0
 #else /* !CONFIG_DEBUG */
 	#define DBG	0	/* for ODM & BTCOEX debug */
